deutsche-gesellschaft-fur-psychologie.csl

```diff
@@ -121,23 +130,27 @@
   </macro>
   <macro name="publisher">
     <choose>
-      <if type="report thesis" match="any">
-        <group delimiter=": ">
-          <text variable="publisher-place"/>
-          <text variable="publisher"/>
-        </group>
-      </if>
-      <else>
+      <if type="article-newspaper" match="none">
         <choose>
-          <if variable="event" match="none">
-            <text variable="genre" suffix=", "/>
+          <if type="report thesis" match="any">
+            <group delimiter=": ">
+              <text variable="publisher-place"/>
+              <text variable="publisher"/>
+            </group>
           </if>
+          <else>
+            <choose>
+              <if variable="event" match="none">
+                <text variable="genre" suffix=", "/>
+              </if>
+            </choose>
+            <group delimiter=": ">
+              <text variable="publisher-place"/>
+              <text variable="publisher"/>
+            </group>
+          </else>
         </choose>
```

Comment:
In this case, this construct is due to APA's differing rules for genre/medium for reports and theses compared to other types. This choose is to avoid printing genre twice for these two types.